src/core/server/plugins/plugins_service.ts

```diff
@@ -336,6 +336,18 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
         }
       }
 
+      // validate that OSS plugins do not have required dependencies on X-Pack plugins
+      if (plugin.source === 'oss') {
```

Comment:
optional: the validation path is growing too large. @elastic/kibana-core we need to move it under a dedicated method to simplify reading.

Reply:
Done in b6a38a72